PurchasesCoreSwift/Public/Purchases.swift

```diff
@@ -0,0 +1,1275 @@
+
+/**
+ * `Purchases` is the entry point for Purchases.framework. It should be instantiated as soon as your app has a unique
+ * user id for your user. This can be when a user logs in if you have accounts or on launch if you can generate a random
+ * user identifier.
+ *  @warning Only one instance of Purchases should be instantiated at a time! Use a configure method to let the
+ *  framework handle the singleton instance for you.
+ */
+@objc(RCPurchases) public class Purchases: NSObject {
+
+    /**
+     * Delegate for `Purchases` instance. The delegate is responsible for handling promotional product purchases and
+     * changes to purchaser information.
+     */
+    @objc public var delegate: PurchasesDelegate? {
+        get { privateDelegate }
+        set {
+            privateDelegate = newValue
+            purchaserInfoManager.delegate = self
+            purchaserInfoManager.sendCachedPurchaserInfoIfAvailable(appUserID: appUserID)
+            Logger.debug(Strings.configure.delegate_set)
+        }
+    }
+
+    let subscriberAttributesManager: SubscriberAttributesManager
+
+    private weak var privateDelegate: PurchasesDelegate?
+    private let operationDispatcher: OperationDispatcher
+
+    /**
+     * Enable automatic collection of Apple Search Ads attribution. Disabled by default
+     */
+    @objc static var automaticAppleSearchAdsAttributionCollection: Bool = false
+
+    /**
+     * Used to set the log level. Useful for debugging issues with the lovely team @RevenueCat
+     */
+    @objc static var logLevel: LogLevel {
+        get { Logger.logLevel }
+        set { Logger.logLevel = newValue }
+    }
+
+    /**
+     * Set this property to your proxy URL before configuring Purchases *only* if you've received a proxy key value
+     * from your RevenueCat contact.
+     */
+    @objc static var proxyURL: URL? {
+        get { SystemInfo.proxyURL }
+        set { SystemInfo.proxyURL = newValue }
+    }
+
+    /**
+     * Set this property to true *only* if you're transitioning an existing Mac app from the Legacy
+     * Mac App Store into the Universal Store, and you've configured your RevenueCat app accordingly.
+     * Contact support before using this.
+     */
+    @objc static var forceUniversalAppStore: Bool {
+        get { SystemInfo.forceUniversalAppStore }
+        set { SystemInfo.forceUniversalAppStore = newValue }
+    }
+
+    /**
+     * Set this property to true *only* when testing the ask-to-buy / SCA purchases flow. More information:
+     * http://errors.rev.cat/ask-to-buy
+     */
+    @available(iOS 8.0, macOS 10.14, watchOS 6.2, macCatalyst 13.0, *)
+    @objc static var simulatesAskToBuyInSandbox: Bool {
+        get { StoreKitWrapper.simulatesAskToBuyInSandbox }
+        set { StoreKitWrapper.simulatesAskToBuyInSandbox = newValue }
+    }
+
+    /**
+     * Enable debug logging. Useful for debugging issues with the lovely team @RevenueCat.
+     */
+    @objc static var debugLogsEnabled: Bool {
```

Comment:
i thought this got deprecated?

Reply:
good catch, we should add a [deprecation notice here](https://www.hackingwithswift.com/example-code/language/how-to-use-available-to-deprecate-old-apis)

Reply:
Should we remove this now?

Reply:
well, we're doing a major... but then again, this was deprecated pretty recently [(July 15th)](https://github.com/RevenueCat/purchases-ios/releases/tag/3.12.0) and it internally redirects to `Logger`, so having it around a bit longer shouldn't be a problem, so I'd vote for keeping it.
It was recent enough that I think even a few of our sample apps + our hybrid SDKs are still using the old one

Reply:
👍